src/payment/strategies/laybuy/laybuy-payment-strategy.spec.ts

```diff
@@ -0,0 +1,167 @@
+            const output = await strategy.execute(getOrderRequestBody());
+
+            expect(output).toEqual(store.getState());
+        });
+
+        it('returns checkout state with undefined payload', () => {
+            const payload = {
+                payment: undefined,
+            };
+            expect(() => strategy.execute(payload))
+                .toThrow(PaymentArgumentInvalidError);
+        });
+
+        it('redirect to Laybuy if additional action is required', async () => {
+            const error = new RequestError(getResponse({
+                ...getErrorPaymentResponseBody(),
+                errors: [
+                    { code: 'additional_action_required' },
+                ],
+                provider_data: {
+                    redirect_url: 'https://sandbox-payment.laybuy..com',
+                },
+                status: 'error',
+            }));
+
+            jest.spyOn(paymentActionCreator, 'submitPayment')
+                .mockReturnValue(of(createErrorAction(PaymentActionType.SubmitPaymentFailed, error)));
+
+            strategy.execute(getOrderRequestBody());
+
+            await new Promise(resolve => process.nextTick(resolve));
+
+            expect(formPoster.postForm).toHaveBeenCalledWith('https://sandbox-payment.laybuy..com', {});
+        });
+
+        it('reject payment when error is different to additional_action_required', async () => {
+            const response = new RequestError(getResponse(getErrorPaymentResponseBody()));
+
+            jest.spyOn(paymentActionCreator, 'submitPayment')
+                .mockReturnValue(of(createErrorAction(PaymentActionType.SubmitPaymentFailed, response)));
+
+            try {
+                await strategy.execute(getOrderRequestBody());
+            } catch (error) {
+                expect(error).toBeInstanceOf(RequestError);
+                expect(formPoster.postForm).not.toHaveBeenCalled();
+            }
+        });
+
+        it('submits payment separately', async () => {
+            const payload = getOrderRequestBody();
+            const options = { methodId: 'laybuy' };
+            await strategy.execute(payload, options);
+
+            expect(paymentActionCreator.submitPayment).toHaveBeenCalledWith(payload.payment);
+            expect(store.dispatch).toHaveBeenCalledWith(submitPaymentAction);
+        });
+    });
+
+    describe('#initialize()', () => {
+        it('initialize strategy', async () => {
+            expect(await strategy.initialize()).toEqual(store.getState());
+        });
+    });
+
+    describe('#finalize()', () => {
+        it('throws an error to inform that order finalization is not required', async () => {
+            const promise = strategy.finalize();
+
+            return expect(promise).rejects.toBeInstanceOf(OrderFinalizationNotRequiredError);
+        });
+    });
+
+    describe('#deinitialize()', () => {
+        it('deinitializes strategy', async () => {
```

Comment:
it returns the state? it's a good practice to make the title of the test match what's being asserted